src/storage/src/source/postgres/snapshot.rs

```diff
@@ -330,38 +330,40 @@ pub(crate) fn render<G: Scope<Timestamp = MzOffset>>(
                 use_snapshot(&client, &snapshot).await?;
             }
 
-            let upstream_info = match mz_postgres_util::publication_info(
-                &config.config.connection_context.ssh_tunnel_manager,
-                &connection_config,
-                &connection.publication,
-            )
-            .await
-            {
-                // If the replication stream cannot be obtained in a definite way there is
-                // nothing else to do. These errors are not retractable.
-                Err(PostgresError::PublicationMissing(publication)) => {
-                    let err = DefiniteError::PublicationDropped(publication);
-                    for oid in reader_snapshot_table_info.keys() {
-                        // Produce a definite error here and then exit to ensure
-                        // a missing publication doesn't generate a transient
-                        // error and restart this dataflow indefinitely.
-                        //
-                        // We pick `u64::MAX` as the LSN which will (in
-                        // practice) never conflict any previously revealed
-                        // portions of the TVC.
-                        let update = ((*oid, Err(err.clone().into())), MzOffset::from(u64::MAX), 1);
-                        raw_handle.give_fueled(&data_cap_set[0], update).await;
+            let client = connection_config
```

Comment:
This client shadows the client previously established which has entered a particular isolation snapshot to get a consistent view of the tables. Also I think this client connection will linger until we finish snapshotting so we probably want to drop it earlier when we're done with it.

Reply:
oof, good catch!

Reply:
is there any reason why I shouldnt use the existing `client` for querying the publication info too?

Reply:
It looks like reusing it caused a bunch of tests to fail with timeout related logic, likely since that client has a special statement timeout set on it. I've reverted back to using a separate client here and dropping it after we get the publication schemas

Reply:
> is there any reason why I shouldnt use the existing client for querying the publication info too?

One of the workers puts its client in replication mode which only allows simple queries and I think we do non-simple queries in the methods called.